test/wait-task/pkg/reconciler/reconciler_test.go

```diff
@@ -0,0 +1,166 @@
+/*
+Copyright 2021 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package reconciler
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"knative.dev/pkg/apis"
+	"knative.dev/pkg/controller"
+)
+
+var validRef = &v1alpha1.TaskRef{
+	APIVersion: "wait.testing.tekton.dev/v1alpha1",
+	Kind:       "Wait",
+}
+
+func TestReconcile(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
+	r := &v1alpha1.Run{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "run",
+		},
+		Spec: v1alpha1.RunSpec{
+			Ref: validRef,
+			Params: []v1beta1.Param{{
+				Name:  "duration",
+				Value: *v1beta1.NewArrayOrString("1s"),
+			}},
+		},
+	}
```

Comment:
let's use yaml helpers here and the rest of this file

Reply:
Oh need extra confirmation here. The yaml helpers is outside of this wait-task/ module, if we want to use it, need to copy paste related funcs in test/parse/yaml.go to somewhere in this wait-task module, is that what you mean?

Reply:
ah yea, you're right - let's not copy paste it